### PR TITLE
REFACTOR: Extend consistency on plot methods

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
Following https://github.com/ansys/pyaedt/pull/4626 and https://github.com/ansys/pyaedt/pull/4708, theses changes aim at making our code base consistent.

The changes consist in returning `matplotlib` figure objects on methods that were returning a list of value.
In the case of `polar_plot_3d_pyvista`, a `pyvista` plotter object is returned when, before, a boolean was returned if one wanted to show the figure or save it in a file.

This changes should be performed before merging #4702

### CODE REVIEW

File: pyaedt/modules/solutions.py [hunk 1]
```diff
@@ -938,8 +938,8 @@ def plot_3d(
 
         Returns
         -------
-        :class:`matplotlib.plt`
-            Matplotlib fig object.
+        :class:`matplotlib.figure.Figure`
```
Comment (dcrawforAtAnsys) [inline pyaedt/modules/solutions.py:941]:
```suggestion
        :class:`matplotlib.pyplot.Figure`
```

File: pyaedt/modules/solutions.py [hunk 2]
```diff
@@ -1849,30 +1847,29 @@ def plot_2d_cut(
...
-            else:
-                return plot_2d_chart(
-                    curves,
-                    xlabel=x_key,
-                    ylabel=quantity,
-                    title=title,
-                    snapshot_path=image_path,
-                    show_legend=show_legend,
-                )
+        show_legend = show
```
Comment (dcrawforAtAnsys) [inline pyaedt/modules/solutions.py:1850]: show_legend is different from show so I don't think show_legend should be set to show. They are two different things.

File: pyaedt/modules/solutions.py [hunk 3]
```diff
@@ -1913,14 +1910,12 @@ def polar_plot_3d(
             Full path for the image file. The default is ``None``, in which case a file is not exported.
         show : bool, optional
             Whether to show the plot. The default is ``True``.
-            If ``False``, the Matplotlib instance of the plot is shown.
+            If ``False``, the Matplotlib instance of the plot is not shown.
```
Comment (dcrawforAtAnsys) [inline pyaedt/modules/solutions.py:1913]:
```suggestion
            If ``False``, the ``plt.Figure`` instance of the plot is not shown.
```
Just semantics, but it is the figure rendering that in changed by `show`

**Review by dcrawforAtAnsys — APPROVED** [review]
See minor suggestions, otherwise looks good.

File: pyaedt/modules/solutions.py (hunk 2 above)
Comment (SMoraisAnsys) [reply pyaedt/modules/solutions.py:1852]: I do agree and, as a matter of fact, I just wanted to follow previous logic (if show then show_legend = True).
I'll update it such that they are different things

File: pyaedt/modules/solutions.py (hunk 1 above)
Comment (SMoraisAnsys) [reply pyaedt/modules/solutions.py:943]: If you look at the source code of matplotlib, matplotlib.pyplot imports Figure from matplotlib.figure.
Do you still want to keep this change though ?

File: pyaedt/modules/solutions.py (hunk 3 above)
Comment (SMoraisAnsys) [reply pyaedt/modules/solutions.py:1918]: Thanks for the remark ! I'll handle this comment depending on your answer to my previous question.

File: pyaedt/modules/solutions.py (hunk 1 above)
Comment (dcrawforAtAnsys) [reply pyaedt/modules/solutions.py:943]: @SMoraisAnsys : matplotlib.figure.Figure is correct. You can ignore my comments.

File: pyaedt/modules/solutions.py (hunk 3 above)
Comment (dcrawforAtAnsys) [reply pyaedt/modules/solutions.py:1918]: @SMoraisAnsys : It wasn't clear to me what the ``pyplot`` interface was doing. Now it's clear. It is the Figure instance that is rendered, and ``pyplot`` is just along for the ride to do the bookkeeping.

**Comment by dcrawforAtAnsys** [comment]
All good.

**Review by dcrawforAtAnsys — COMMENTED** [review]
LGTM.

**Comment by dcrawforAtAnsys** [comment]
Inadvertently closed. LGTM.

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Pull request merged**